File: trunk/C#/RBD/RBD/Msg/Client/ResultSetMessage.cs
// +

using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

using RBD.Communication;
using RBD.DB;

namespace RBD.Msg.Client
{
    public class ResultsetMessage : SuccessMessage
    {
        private String result;

        public void setResultSet(DatabaseTable table)
        {
            this.setResult(table.ToString());
        }

        public void setResult(String result)
        {
            this.result = result;
        }

        public String getResult()
        {
            return result;
        }

        override
        public void FromBinary(DataInputStream s) //throws IOException 
        {
            result = s.ReadString();
        }

        override
        public MessageType GetMessageType()
        {
            return MessageType.CLIENT_RESULTSET;
        }

        override
        public void ToBinary(DataOutputStream s) //throws IOException 
        {
            s.WriteString(result);
        }
    }
}

File: SR-RBD/src/ddb/msg/client/ResultsetMessage.java
/**
 * 
 */
package ddb.msg.client;

import java.io.IOException;

import ddb.communication.DataInputStream;
import ddb.communication.DataOutputStream;
import ddb.db.DatabaseTable;
import ddb.msg.MessageType;

/** 
 * <!-- begin-UML-doc -->
 * <!-- end-UML-doc -->
 * @author User
 * @generated "UML to Java (com.ibm.xtools.transform.uml2.java5.internal.UML2JavaTransform)"
 */
public class ResultsetMessage extends SuccessMessage {
	/** 
	 * <!-- begin-UML-doc -->
	 * <!-- end-UML-doc -->
	 * @generated "UML to Java (com.ibm.xtools.transform.uml2.java5.internal.UML2JavaTransform)"
	 */
	private String result;

	/** 
	 * <!-- begin-UML-doc -->
	 * <!-- end-UML-doc -->
	 * @param resultSet
	 * @generated "UML to Java (com.ibm.xtools.transform.uml2.java5.internal.UML2JavaTransform)"
	 */
	public void setResultSet(DatabaseTable table) {
		this.setResult(table.toString());
	}
	
	public void setResult(String result) {
		this.result = result;
	}

	public String getResult() {
		return result;
	}
	
	@Override
	public void fromBinary(DataInputStream s) throws IOException {
		setResult(s.readString());
	}

	@Override
	public MessageType getType() {
		return MessageType.CLIENT_RESULTSET;
	}

	@Override
	public void toBinary(DataOutputStream s) throws IOException {
		s.writeString(getResult());
	}
}